server/src/main/java/com/cloud/resource/ResourceManagerImpl.java

```diff
@@ -2746,6 +2753,13 @@ public HostVO fillRoutingHostVO(final HostVO host, final StartupRoutingCommand s
             throw new IllegalArgumentException("Can't add host whose hypervisor type is: " + hyType + " into cluster: " + clusterVO.getId() +
                     " whose hypervisor type is: " + clusterVO.getHypervisorType());
         }
+        CPU.CPUArchitecture hostCpuArchitecture = CPU.CPUArchitecture.fromType(ssCmd.getCpuArchitecture());
+        if (hostCpuArchitecture != null && clusterVO.getArch() != null && hostCpuArchitecture != clusterVO.getArch()) {
+            String msg = String.format("Can't add a host whose architecture is: %s into cluster of architecture type: %s",
+                    hostCpuArchitecture.getType(), clusterVO.getArch().getType());
+            logger.error(msg);
+            throw new IllegalArgumentException(msg);
+        }
```

Comment:
this bit makes the isMultiArch flag necessary. I think we shouldn't implement that and allow dynamic use of architectures (responsibility of the operator) cc @rajujith @andrijapanicsb @NuxRo

Reply:
The multiArch flag is a dynamic check, whenever you add a different arch cluster in your existing zone, then the multiArch flag becomes true and allows the operator to select the template arch

Reply:
So what if all hosts/clusters for a certain arch are removed, will it fall back to single arch? (i.e. what is the use of the flag)

Reply:
The flag is used for the UI to display the different archs available so users can filter templates by arch on the VM deployment wizard

Reply:
again, why is that not always available? I still don't see the need for the flag. a zone has hosts and hosts can only be grouped with hosts of the same arch in a cluster, but there is no reason to assume a zone is of only one arch at any time, is there?